lib/ui/window/window.cc

```diff
@@ -425,6 +425,50 @@ void Window::CompletePlatformMessageResponse(int response_id,
   response->Complete(std::make_unique<fml::DataMapping>(std::move(data)));
 }
 
+Dart_Handle ComputePlatformResolvedLocale(Dart_Handle supportedLocalesHandle) {
+  intptr_t length;
+  Dart_ListLength(supportedLocalesHandle, &length);
+  size_t u_length = static_cast<size_t>(length);
+  std::vector<std::string> supportedLocales;
+  for (size_t i = 0; i < u_length; i++) {
+    const char* cname = nullptr;
+    Dart_Handle element = Dart_ListGetAt(supportedLocalesHandle, i);
+    if (Dart_IsNull(element)) {
+      supportedLocales.emplace_back("");
+      continue;
+    }
+    Dart_StringToCString(Dart_ListGetAt(supportedLocalesHandle, i), &cname);
+    supportedLocales.emplace_back(cname);
+  }
+  std::vector<std::string> results =
+      *UIDartState::Current()
+           ->window()
+           ->client()
+           ->ComputePlatformResolvedLocale(supportedLocales);
+
+  if (results.empty()) {
+    return Dart_Null();
```

Comment:
There is an inconsistency about whether an empty result is represented as an empty list or as a null.

The C++ `PlatformView::ComputePlatformResolvedLocales` and Java `FlutterJNI.computePlatformResolvedLocale` methods return an empty array if no locale is found.  But the Dart native `_computePlatformResolvedLocale` will return null.

It might be simplest to use a non-null list of strings everywhere.